chromium_src/content/public/common/content_features.cc

```diff
@@ -10,14 +10,20 @@
 
 namespace features {
 
+// This is intended as a kill switch for the Idle Detection feature. To enable
+// this feature, the experimental web platform features flag should be set,
+// or the site should obtain an Origin Trial token.
+BASE_FEATURE(kIdleDetection,
```

Comment:
@goodov This change is still marked as WIP, because it needs to be finished with some input from reviewers. The main issue here is that `IdleDetection` has been enabled by default upstream, and security at brave still wants us to disable it by default. What complicates things a bit further is that we do have a switch on our setting to allow users to toggle this feature on and off, however the mechanism for this setting to be communicated from content to blink has been gutted by upstream [1].

I had added this code with the intention of revisiting it, and eventually introduce an override on blink [2] to permit us to keep this feature off if necessary. However, I'm not sure if that would be possible easily to check for the status of `kIdleDetection` from blink.

[1] https://chromium.googlesource.com/chromium/src/+/90b6c7b5a926e8a8c555a720cb2054ba6f4e9cc3
[2] https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/idle/idle_detector.cc;l=119

Reply:
Also related to this is the change in bac0f3de3112382e8cce3c96ddca40e0c31dc685 which makes `IDLE_DETECTION` `BLOCKED` by default. The feature flag basically hides the related Settings UI elements, while content settings change should prevent the permissions dialog from popping up.